src/main/java/org/kohsuke/github/GHGistUpdater.java

```diff
@@ -0,0 +1,60 @@
+package org.kohsuke.github;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+
+/**
+ * Builder pattern for updating a Gist.
+ *
+ * @author Martin van Zijl
+ */
+public class GHGistUpdater {
+    private final GHGist base;
+    private final Requester builder;
+    LinkedHashMap<String,Object> files;
+
+    GHGistUpdater(GHGist base) {
+        this.base = base;
+        this.builder = new Requester(base.root);
+
+        files = new LinkedHashMap<>();
+    }
+
+    public GHGistUpdater addFile(String fileName, String content) throws IOException {
+        updateFile(fileName, content);
+        return this;
+    }
+
+//    // This method does not work.
+//    public GHGistUpdater deleteFile(String fileName) throws IOException {
```

Comment:
@martinvanzijl 
This probably has to do with the way that values are passed via maps.  
I suspect passing `null` results in the key not being set or at least not set to GitHub.